hbase-server/src/main/java/org/apache/hadoop/hbase/replication/HBaseReplicationEndpoint.java

```diff
@@ -295,6 +296,7 @@ private synchronized AsyncClusterConnection getPeerConnection() throws IOExcepti
           .createBlockingRpcChannel(master, User.getCurrent(), shortOperationTimeout));
       ListReplicationSinkServersResponse resp = masterStub
         .listReplicationSinkServers(null, ListReplicationSinkServersRequest.newBuilder().build());
+      isReplicationServer = resp.getIsReplicationServer();
```

Comment:
Do you have to do hasReplicationServer first? And then read it if present? Or maybe this defaults false if not present?

Reply:
This field is required. So not need to check hasReplicationServer first?